SConstruct

```diff
@@ -222,6 +222,7 @@ opts.Add(BoolVariable("xaudio2", "Enable the XAudio2 audio driver", False))
 opts.Add(BoolVariable("vulkan", "Enable the vulkan rendering driver", True))
 opts.Add(BoolVariable("opengl3", "Enable the OpenGL/GLES3 rendering driver", True))
 opts.Add(BoolVariable("d3d12", "Enable the Direct3D 12 rendering driver", False))
+opts.Add(BoolVariable("metal", "Enable the Metal rendering driver (Apple arm64 only)", False))
```

Comment:
We should consider enabling `metal` by default sooner than later IMO, as it should probably be our preferred RenderingDevice implementation for macOS and iOS.

`d3d12` above defaults to False because it adds some build dependencies and we also used to have to ship a proprietary DLL for it (now solved), but I guess for Metal that may not be a concern, all deps should be part of the macOS and iOS SDKs?

Reply:
Hm given later considerations on the lack of support for `metal` on non-Apple GPUs, this should probably stay opt-in for now indeed.

I wonder whether it will work for official `universal` builds with `arm64` and `x86_64` lipo'd together?

Reply:
@akien-mga yes, it should work with lipo'd builds, as they have no dependencies on each other.

Perhaps we can enable dynamically at build time, such that ARM builds have Metal by default?

Reply:
IMO ARM builds should have metal onby default while X86 builds should have it off by default

Reply:
I agree – best way to start getting it tested 👍🏻